sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/pipeline_component.py

```diff
@@ -178,13 +178,23 @@ def _validate_compute_is_set(self, *, parent_node_name=None):
             if isinstance(node, BaseNode) and node._skip_required_compute_missing_validation:
                 continue
             if has_attr_safe(node, "compute") and node.compute is None:
-                no_compute_nodes.append(node_name)
+                # node with only resources, which has different behaviour from standalone job;
+                # so special handle them with different error message.
+                if has_attr_safe(node, "resources") and node.resources is not None:
+                    no_compute_with_resources_nodes.append(node_name)
+                else:
+                    no_compute_nodes.append(node_name)
 
         for node_name in no_compute_nodes:
             validation_result.append_error(
                 yaml_path=f"jobs.{parent_node_name}{node_name}.compute",
                 message="Compute not set",
             )
+        for node_name in no_compute_with_resources_nodes:
+            validation_result.append_error(
+                yaml_path=f"jobs.{parent_node_name}{node_name}.compute",
+                message="Invalid resource is set, please set serverless compute or pipeline default compute",
```

Comment:
Don't quite get this, why can we check invalid resource?  

If we can have resources configured in component.

Reply:
This function `_validate_compute_is_set` will only be called when (root) pipeline default compute is not set. So the node(s) in `no_compute_with_resources_nodes` corresponds to combination "null default compute of pipeline + resources compute of step" and it is invalid, so add one error message here.

Reply:
I'm asking this is because user can set other things like resources.shm_size which is not related to compute. 

We might just say user should set compute here.

Reply:
Got your point, will update this error message.